shared/chat/conversation/info-panel/notifications/container.js

```diff
@@ -25,89 +25,127 @@ const serverStateToProps = (notifications: Types.NotificationsState, type: 'desk
   return 'never'
 }
 
+type _StateProps = {
+  channelWide: boolean,
+  conversationIDKey: string,
+  desktop: Types.NotifyType,
+  mobile: Types.NotifyType,
+  muted: boolean,
+  saveState: Types.NotificationSaveState,
+}
+
+// Use conversationIDKey as a signal for whether StateProps is empty.
+type StateProps = _StateProps | {conversationIDKey: void}
+
 const mapStateToProps = (state: TypedState): StateProps => {
   const conversationIDKey = Constants.getSelectedConversation(state)
   if (!conversationIDKey) {
     logger.warn('no selected conversation')
-    return {}
+    return {conversationIDKey: undefined}
   }
   const inbox = Constants.getSelectedInbox(state)
   if (!inbox) {
     logger.warn('no selected inbox')
-    return {}
+    return {conversationIDKey: undefined}
   }
   const notifications = inbox.get('notifications')
   if (!notifications) {
     logger.warn('no notifications')
-    return {}
+    return {conversationIDKey: undefined}
   }
   const desktop = serverStateToProps(notifications, 'desktop')
   const mobile = serverStateToProps(notifications, 'mobile')
   const muted = Constants.getMuted(state)
   const {channelWide} = notifications
   const saveState = inbox.get('notificationSaveState')
 
-  return {
+  return ({
```

Comment:
Flow somehow thinks this could fit both cases of the union, although I don't see how it can fit the second case.